models/user/user.go

```diff
@@ -1283,6 +1283,9 @@ func IsUserVisibleToViewer(ctx context.Context, u, viewer *User) bool {
 		if viewer == nil || viewer.IsRestricted {
 			return false
 		}
+		if viewer.ID == u.ID {
```

Comment:
There is also the problem when u.Visibility is VisibleTypeLimited. So I think put it in line 1270 is better, just like I did in https://github.com/go-gitea/gitea/pull/21211/files